zerver/actions/user_topics.py

```diff
@@ -52,10 +56,10 @@ def do_unmute_topic(
     user_profile: UserProfile, stream: Stream, topic: str, *, skip_muted_topics_event: bool = False
 ) -> None:
     # Note: If you add any new code to this function, the
-    # remove_topic_mute call in do_update_message will need to be
+    # remove_topic_visibility_policy call in do_update_message will need to be
     # updated for correctness.
     try:
-        remove_topic_mute(user_profile, stream.id, topic)
+        remove_topic_visibility_policy(user_profile, stream.id, topic)
```

Comment:
For this one, it's possible that we will eventually want a common code path where you just call `add_topic_visibility_policy`, with a `visibility_policy` of `VISIBILITY_POLICY_INHERIT`; but we can leave that sort of refactor for a future PR.

Reply:
After [CZO discussion](https://chat.zulip.org/#narrow/stream/3-backend/topic/Follow.20Topics/near/1501281),

It is implemented in this PR itself, `set_user_topic_visibility_policy` is the common function.